component/org.wso2.carbon.identity.casque.authenticator/src/test/java/org.wso2.carbon.identity.casque.authenticator/authenticator/CasqueAuthenticatorTest.java

```diff
@@ -0,0 +1,310 @@
+import org.wso2.carbon.identity.casque.authenticator.authenticator.radius.RadiusResponse;
+import org.wso2.carbon.identity.casque.authenticator.constants.CasqueAuthenticatorConstants;
+import org.wso2.carbon.identity.application.authentication.framework.AuthenticatorFlowStatus;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.casque.authenticator.authenticator.internal.CasqueAuthenticatorServiceDataHolder;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+@PrepareForTest({CasqueAuthenticatorServiceDataHolder.class, IdentityTenantUtil.class, RadiusResponse.class,
+        Radius.class, AuthenticatedUser.class, User.class})
+
+public class CasqueAuthenticatorTest {
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    private CasqueAuthenticator casqueAuthenticator;
+
+    @Mock
+    private HttpServletRequest httpServletRequest;
+
+    @Mock
+    private HttpServletResponse httpServletResponse;
+
+    @Mock
+    private UserStoreManager userStoreManager;
+
+    @Mock
+    CasqueAuthenticatorServiceDataHolder casqueAuthenticatorServiceDataHolder;
+
+    @Mock
+    RealmService realmService;
+
+    @Mock
+    UserRealm tenantUserRealm;
+
+    @Mock
+    private AuthenticationContext context;
+
+    @Mock
+    private RadiusResponse radiusResponse;
+
+    @Mock
+    private AuthenticatedUser authenticatedUser;
+
+    @Mock
+    private User user;
+
+    @Mock
+    private Map mockMap;
+
+    @BeforeMethod
+    public void setUp() {
+
+        casqueAuthenticator = new CasqueAuthenticator();
+        initMocks(this);
+    }
+
+    @Test(description = "Test case for GetName() method.")
+    public void testGetName() {
+
+        Assert.assertEquals(casqueAuthenticator.getName(),
+                CasqueAuthenticatorConstants.AUTHENTICATOR_NAME);
+    }
+
+    @Test(description = "Test case for RetryAuthenticationEnabled() method.")
+    protected void testRetryAuthenticationEnabled() {
+
+        Assert.assertFalse(casqueAuthenticator.retryAuthenticationEnabled());
+    }
+
+    @Test(description = "Test case for GetFriendlyName() method.")
+    public void testGetFriendlyName() {
+
+        Assert.assertEquals(casqueAuthenticator.getFriendlyName(),
+                CasqueAuthenticatorConstants.AUTHENTICATOR_FRIENDLY_NAME);
```

Comment:
Shall we move this to the above line, if length does not exceed 120?

Reply:
It exceed 120